### PR TITLE
feat(watch): add a watch verb

### PR DESCRIPTION
**Pull request by ofrobots**
Fixes: https://github.com/google/ts-style/issues/81

WIP but opening now to get early feedback.  

Still to do:
- [ ] implement tests
- [ ] child process error handling
- [ ] avoid running a then/else command when a previous one has not completed.

### CODE REVIEW

File: src/watch.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    return ret;
+  };
+
+  ts.createWatchProgram(host);
+  return true;
+}
+
+function reportDiagnostic(diagnostic: ts.Diagnostic) {
+  console.error(
```
Comment (kjin) [inline src/watch.ts]: Use `options.logger` here?

**Comment by carnesen** [comment]
I see now that this project is more of a multi-purpose ts dev cli, not limited to just style and formatting 👍 . From the description of what the `watch` command does, it would be more consistent with other CLI tools to put that functionality under `gts compile --watch`... I would expect `gts watch` just to watch. So if I wanted to compile and test on change I'd do `gts watch --then 'npm run compile && npm run test`. Or maybe `gts watch --on-change 'npm run compile && npm run test`.

**Comment by ofrobots** [comment]
> I would expect gts watch just to watch

I am not sure if I agree with this. AFAIU `tsc -w` doesn't just watch, it runs forever and triggers incremental recompilation. Ideally `tsc` should have some hooks to run commands after each {un}successful incremental compilation, but it doesn't. `gts` is stepping in to provide this missing functionality.

> So if I wanted to compile and test on change I'd do gts watch --then 'npm run compile && npm run test.

This would be suboptimal. We would want an *incremental* compilation when using `watch`. `npm run compile` will force a full compilation. Right now the only way to cause incremental compilation (AFAICT) is programmatically. We need a `gts` verb to do incremental compilation, and IMO `watch` matches my intuition the most here.

**Comment by carnesen** [comment]
@ofrobots I don't want to say your intuition is wrong, but I'd argue it's inconsistent with the other CLI APIs in this space (ava, tsc, mocha, tslint, eslint, jest), for which the user provides a `--watch` flag to enable an incremental/watcher mode. If the proposed "watch" command's primary job is to do incremental compilation, `gts compile --watch` would be more intuitive to more users. As you said, that's `tsc`'s API! 

If it's just `gts watch`, it's unclear to the user whether that's enabling a watch mode for `fix`, `check`, `compile` or something else. As I mentioned previously, I'd actually guess that a `watch` command would actually do just and only that (watch) similar to [chokidar-cli](https://github.com/kimmobrunfeldt/chokidar-cli).

**Comment by ofrobots** [comment]
I'm going to abandon this until I can get my head around a clear value proposition.

**Pull request closed without merging**